scipy/sparse/linalg/eigen/_svds.py

```diff
@@ -163,45 +237,28 @@ def svds(A, k=6, ncv=None, tol=0, which='LM', v0=None,
     True
 
     """
-    if which == 'LM':
-        largest = True
-    elif which == 'SM':
-        largest = False
-    else:
-        raise ValueError("which must be either 'LM' or 'SM'.")
 
-    if not (isinstance(A, LinearOperator) or isspmatrix(A)
-            or is_pydata_spmatrix(A)):
-        A = np.asarray(A)
+    args = _iv(A, k, ncv, tol, which, v0, maxiter, return_singular_vectors,
+               solver)
+    A, k, ncv, tol, which, v0, maxiter, return_singular_vectors, solver = args
 
+    largest = (which == 'LM')
     n, m = A.shape
```

Comment:
Is this where `svds` doesn't match docs?  I would have expected `m, n = A.shape`

Reply:
No, surprisingly. Despite flipping `n, m` between code and documentation, everything is right except for the inequality strictness being swapped in the documentation.